common/txmgr/types/chain.go

```diff
@@ -4,7 +4,10 @@ import "fmt"
 
 // Sequence represents the base type, for any chain's sequence object.
 // It should be convertible to a string
-type Sequence fmt.Stringer
+type Sequence interface {
+	fmt.Stringer
+	Int64() int64
```

Comment:
nit: Could you add comment why this is needed? For example, needed till we remove the nonce comparison from txmgr code.